internal/p2p/peermanager.go

```diff
@@ -867,6 +875,22 @@ func (m *PeerManager) Disconnected(ctx context.Context, peerID types.NodeID) {
 	delete(m.evicting, peerID)
 	delete(m.ready, peerID)
 
+	if peer, ok := m.store.Get(peerID); ok {
+		peer.LastDisconnected = time.Now()
+		_ = m.store.Set(peer)
+		// launch a thread to ping the dialWaker when the
+		// disconnected peer can be dialed again.
+		go func() {
+			timer := time.NewTimer(time.Until(time.Now().Add(m.options.DisconnectCooldownPeriod)))
```

Comment:
Can this not just be the duration directly? Any reason to do the until...add stuff?

Reply:
oh, good call